src/engine/engine_wt.c

```diff
@@ -32,14 +110,18 @@ static void _ocf_write_wt_req_complete(struct ocf_request *req)
 		req->complete(req, req->info.core_error ? req->error : 0);
 
 		ocf_engine_invalidate(req);
+
+		return;
+	}
+
+	if (req->info.flush_metadata) {
```

Comment:
I think this gets set in _ocf_write_wt_update_bit, so will always by false here. you should change this condition to req->info.dirty_any